openfl/transport/grpc/director_client.py

```diff
@@ -0,0 +1,280 @@
+                if response.size == len(response.npbytes):
+                    content_file.write(response.npbytes)
+                else:
+                    raise Exception('Broken archive')
+
+        shutil.unpack_archive(arch_name, experiment_name)
+        os.remove(arch_name)
+
+        requirements_filename = f'./{experiment_name}/requirements.txt'
+
+        if os.path.isfile(requirements_filename):
+            attempts = 3
+            for _ in range(attempts):
+                try:
+                    check_call([
+                        executable, '-m', 'pip', 'install', '-r', requirements_filename],
+                        shell=False)
+                except Exception as exc:
+                    logger.error(f'Failed to install requirements: {exc}')
+                    time.sleep(3)
+        else:
+            logger.error('No ' + requirements_filename + ' file found.')
+
+    def _get_experiment_data(self):
+        """Generate the experiment data request."""
+        yield director_pb2.WaitExperimentRequest(collaborator_name=self.shard_name)
+
+    def _get_node_info(self):
+        """Generate a node info message."""
+        return director_pb2.NodeInfo(name=self.shard_name)
+
+    def send_health_check(self, collaborator_name, is_experiment_running, valid_duration):
+        """Send envoy health check."""
+        status = director_pb2.CollaboratorStatus(
+            name=collaborator_name,
+            is_experiment_running=is_experiment_running,
+        )
+        status.valid_duration.seconds = valid_duration
+        logger.debug(f'Sending health check status: {status}')
+
+        return self.stub.CollaboratorHealthCheck(status)
+
+
+class DirectorClient:
+    """Director client class for users."""
+
+    def __init__(self, client_id, director_uri, disable_tls, root_ca, key, cert) -> None:
+        """Initialize director client object."""
+        channel_opt = [('grpc.max_send_message_length', 512 * 1024 * 1024),
+                       ('grpc.max_receive_message_length', 512 * 1024 * 1024)]
+        if disable_tls:
+            channel = grpc.insecure_channel(director_uri, options=channel_opt)
+        else:
+            with open(root_ca, 'rb') as f:
+                root_ca_b = f.read()
+            with open(key, 'rb') as f:
+                key_b = f.read()
+            with open(cert, 'rb') as f:
+                cert_b = f.read()
+
+            credentials = grpc.ssl_channel_credentials(
+                root_certificates=root_ca_b,
+                private_key=key_b,
+                certificate_chain=cert_b
+            )
+
+            channel = grpc.secure_channel(director_uri, credentials, options=channel_opt)
+        self.stub = director_pb2_grpc.FederationDirectorStub(channel)
+
+        self.client_id = client_id
+        self.header = director_pb2.RequestHeader(sender=self.client_id)
+
+    def report_shard_info(self, shard_descriptor) -> bool:
```

Comment:
This method was probably put here by mistake and should be removed

Reply:
Removed